File: server-v1/src/db/models/post.model.js
const { Model, DataTypes, Sequelize } = require("sequelize");
const POST_TABLE = "post";

const PostSchema = {
  id: {
    allowNull: false,
    autoIncrement: true,
    primaryKey: true,
    type: DataTypes.INTEGER,
  },
  title: {
    allowNull: false,
    type: DataTypes.STRING,
  },
  content: {
    allowNull: false,
    type: DataTypes.TEXT,
  },
  image: {
    allowNull: true,
    type: DataTypes.STRING,
  },
  createdAt: {
    allowNull: false,
    type: DataTypes.DATE,
    defaultValue: Sequelize.NOW,
  },
};

class Post extends Model {
  static associate(models) {
    // Relación 1:N entre User y Post
    this.belongsTo(models.User, {
      foreignKey: "userId",
      as: "user",
    });
  }
  static config(sequelize) {
    return {
      sequelize,
      tableName: POST_TABLE,
      modelName: "Post",
      timestamps: false,
    };
  }
}

module.exports = { POST_TABLE, PostSchema, Post };

File: server-v2/src/models/post_model.py
from datetime import datetime
import config.db as db

db = db.db

class Post(db.Model):
  id = db.Column(db.Integer, primary_key=True, autoincrement=True)
  title = db.Column(db.String(80), nullable=False)
  content = db.Column(db.Text, nullable=False)
  created_at = db.Column(db.DateTime, nullable=False, default=datetime.utcnow)
  # Relación con la tabla User
  user_id = db.Column(db.Integer, db.ForeignKey('user.id'), nullable=False)

  def __init__(self, title, content, user_id):
    self.title = title
    self.content = content
    self.user_id = user_id
